src/main/java/mskory/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package mskory.bookstore.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mskory.bookstore.dto.order.OrderCreateRequestDto;
+import mskory.bookstore.dto.order.OrderItemResponseDto;
+import mskory.bookstore.dto.order.OrderResponseDto;
+import mskory.bookstore.dto.order.OrderStatusUpdateRequestDto;
+import mskory.bookstore.exception.OrderPlacingException;
+import mskory.bookstore.exception.StatusNotFoundException;
+import mskory.bookstore.mapper.OrderItemMapper;
+import mskory.bookstore.mapper.OrderMapper;
+import mskory.bookstore.model.order.Order;
+import mskory.bookstore.model.order.OrderItem;
+import mskory.bookstore.model.order.Status;
+import mskory.bookstore.model.shoppingcart.ShoppingCart;
+import mskory.bookstore.model.user.User;
+import mskory.bookstore.repository.OrderItemRepository;
+import mskory.bookstore.repository.OrderRepository;
+import mskory.bookstore.service.OrderService;
+import mskory.bookstore.service.ShoppingCartService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Transactional
+    public OrderResponseDto create(OrderCreateRequestDto orderCreateRequestDto) {
```

Comment:
While Override annotation is technically optional it's always nice to see inherited/implemented methods